framework/Objects/Meta_Fields.php

```diff
@@ -0,0 +1,266 @@
+<?php
+
+namespace JustCoded\WP\Framework\Objects;
+
+/**
+ * Meta_Fields is the base class for acquiring data from custom fields.
+ */
+
+class Meta_Fields {
+
+	/**
+	 * Plugin used for extending of post custom fields.
+	 *
+	 * Available values:
+	 *      'just-custom-fields'
+	 *      'advanced-custom-fields'
+	 *
+	 * @var string
+	 */
+	public $custom_fields_plugin = 'just-custom-fields';
+
+	/**
+	 * Current $post object
+	 *
+	 * @var \WP_Post
+	 */
+	public $post;
+
+	/**
+	 * Current $term object
+	 *
+	 * @var \WP_Term
+	 */
+	public $term;
+
+	/**
+	 * Returns the value of an object property.
+	 *
+	 * Do not call this method directly as it is a PHP magic method that
+	 * will be implicitly called when executing `$value = $object->property;`.
+	 *
+	 * @param string $name The property name.
+	 *
+	 * @return mixed the property value
+	 * @throws \Exception Property is not defined.
+	 * @see __set()
+	 */
+	public function __get( $name ) {
+		if ( $this->post ) {
+			// getter for magic property $this->field_*.
+			return $this->get_field( $name );
+		} else {
+			return $this->get_term_field( $name );
+		}
```

Comment:
this is generic class, so we should not check term or post here.

Reply:
deleted